### DISCUSSION TITLE
china ,docker，pull failed，help

### DISCUSSION THREAD

**Discussion by Dalishimin** [teslamate-org/teslamate#3956] [General]
now 1.28.5 ，want to 1.29.1

> nas has gone through the proxy and avoided blocking, but the problem still exists.

docker-compose --version

> Docker Compose version v2.9.0-6413-g38f6acd

In the past, after entering such a command, the upgrade process was automatically completed.
`docker-compose pull`

but，now，i try many times，show：

![屏幕截图 2024-06-15 084400](https://github.com/teslamate-org/teslamate/assets/76463117/b97c61ef-bead-4af6-8647-d3b8548cc7b2)

if i  retry，it show：

> 
[+] Running 0/4
 ⠿ teslamate Error                                                         7.0s
 ⠿ database Error                                                          7.0s
 ⠿ grafana Error                                                           7.0s
 ⠿ mosquitto Error                                                         7.0s
Error response from daemon: Get "https://registry-1.docker.io/v2/": EOF

![微信截图_20240614145532](https://github.com/teslamate-org/teslamate/assets/76463117/c638f5f5-6b42-41ac-8d5c-80317a9e02a5)

tks

**Comment by Dalishimin** [comment]
![微信截图_20240616084134](https://github.com/teslamate-org/teslamate/assets/76463117/1cce693d-d003-44e2-bba2-72bdee30b944)

i retry

**Reply by ShiriNmi1520** [reply]
Maybe it's because the government demands that Docker registry access be [shut down](https://medium.com/@PlanB./chinese-docker-hub-complete-shutdown-how-far-can-kubernetes-image-repositories-go-826706007b8e). It looks like it's even affecting proxy servers.
For your reference, here's the [post](https://juejin.cn/post/7381760851986120743?searchId=20240702114502657BF3E298100164551D) from JueJin to help you migrate the registry image to Aliyun

**Reply by Dalishimin** [reply]
tks i'll try

**Reply by Dalishimin** [reply]
Do you know how to do docker pull to the aliyun image address after aliyun successfully obtains the image?
The usual pull command is like this：
`docker compose pull`

**Reply by Dalishimin** [reply]
hi,i succeed to aliyun,but the last step failed

![image](https://github.com/user-attachments/assets/0def1d34-a5c4-48b0-82f2-9283d1608d42)

**Reply by ShiriNmi1520** [reply]
Can I get a preview of your compose file? It looks like something went wrong with your configuration.

**Reply by Dalishimin** [reply]
> compose file

i am sorry ,i don't know which file do u want.

**Reply by Dalishimin** [reply]
> Can I get a preview of your compose file? It looks like something went wrong with your configuration.

always,the compose file's name like?
i am not a tech,3 years it works fine,pull ok
but from last month pull failed
now the imagine in aliyun
before I don't know how compose file work😓

**Reply by ShiriNmi1520** [reply]
If you have one, the filename should be `docker-compose` with the `yml` or `yaml` extension.
By the way, I saw your `docker pull` command; it looks like your aliyun registry can't find the `teslamate` image with the `latest` tag; can you confirm this on the aliyun site as well?

**Reply by Dalishimin** [reply]
get，tks
the compose file
![image](https://github.com/user-attachments/assets/2c113bc7-2fd6-454e-ba78-76a6e39bbe27)

![image](https://github.com/user-attachments/assets/a5bfb819-e0b0-4b7d-b767-fa963b168dda)

and ,the teslamate image in aliyun
![image](https://github.com/user-attachments/assets/d298348f-98dd-4a44-bf95-d5233518a508)

![image](https://github.com/user-attachments/assets/2ba61f84-4ec2-4b68-b068-6cc79b317342)

![image](https://github.com/user-attachments/assets/2e018c66-c17e-407a-b95d-a0cec015f53b)

**Reply by Dalishimin** [reply]
> get，tks the compose file ![image](https://private-user-images.githubusercontent.com/76463117/351922235-2c113bc7-2fd6-454e-ba78-76a6e39bbe27.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIwNDI1ODcsIm5iZiI6MTcyMjA0MjI4NywicGF0aCI6Ii83NjQ2MzExNy8zNTE5MjIyMzUtMmMxMTNiYzctMmZkNi00NTRlLWJhNzgtNzZhNmUzOWJiZTI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI3VDAxMDQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjNGNjYWRhYzhjNDk3YWNiOGFkN2QyNjMxODFiMGRhYjhkOWU5N2NiMjE2N2Y3NzM3ZGUzZTExMThhMzQ4YWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.nwdc_hE-gihRvTB7NZt6p3E8FHC2hkyBLBtmnRg-Py8)
> 
> ![image](https://private-user-images.githubusercontent.com/76463117/351922434-a5bfb819-e0b0-4b7d-b767-fa963b168dda.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIwNDI1ODcsIm5iZiI6MTcyMjA0MjI4NywicGF0aCI6Ii83NjQ2MzExNy8zNTE5MjI0MzQtYTViZmI4MTktZTBiMC00YjdkLWI3NjctZmE5NjNiMTY4ZGRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI3VDAxMDQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIwZTVmYzdkYjY5NTdmNGZkNjQ4OGU3ZDI5NjM1NzBiZmFkYWNjYWI2ZmFlMDcwYTk4MzgxNzhhNGRiZjEwMjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wIXu7xvLif-nzkIyAHan2DQcq4cWxjBO8RGBs6OxxTs)
> 
> and ,the teslamate image in aliyun ![image](https://private-user-images.githubusercontent.com/76463117/351923028-d298348f-98dd-4a44-bf95-d5233518a508.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIwNDI1ODcsIm5iZiI6MTcyMjA0MjI4NywicGF0aCI6Ii83NjQ2MzExNy8zNTE5MjMwMjgtZDI5ODM0OGYtOThkZC00YTQ0LWJmOTUtZDUyMzM1MThhNTA4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI3VDAxMDQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlmNTRmNDBlNTUwZmU0OTBhYmZhNmI0MTQwNjc0OGFmZmE5ODUxYTU4NjllY2IyMWJmYjE2MGIyNDgyZmEwNzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CP9rkhsYYKJqvXLVyGhNu4Djl2hxWV9JtYk4hUD65pM)
> 
> ![image](https://private-user-images.githubusercontent.com/76463117/351923291-2ba61f84-4ec2-4b68-b068-6cc79b317342.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIwNDI1ODcsIm5iZiI6MTcyMjA0MjI4NywicGF0aCI6Ii83NjQ2MzExNy8zNTE5MjMyOTEtMmJhNjFmODQtNGVjMi00YjY4LWIwNjgtNmNjNzliMzE3MzQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI3VDAxMDQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmYTkwNjczYWJlOTg5N2NjYTg2MTg0NmM1MjJiZTQ0NThhYjY3NjRkMjRjNGQ5MTJkODE4MjlmMWE2YmE0OTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3Xk75rBWgqvcxMCl1hbTFw6IANGVb9j1uWTkH0-6YQg)
> 
> ![image](https://private-user-images.githubusercontent.com/76463117/351923396-2e018c66-c17e-407a-b95d-a0cec015f53b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIwNDI1ODcsIm5iZiI6MTcyMjA0MjI4NywicGF0aCI6Ii83NjQ2MzExNy8zNTE5MjMzOTYtMmUwMThjNjYtYzE3ZS00MDdhLWI5NWQtYTBjZWMwMTVmNTNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI3VDAxMDQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlYTVlNmI3M2FkODhjMWI2Y2JmMGRjZDNhOWZjOTAyZWUwMmM5YTUzYTNjMjExYmJmOWQyOWJhZTQ3ZGMxYzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bcyCueW9mdehLAubegfM9yAkIZiyd0W1oZjvEOfUfbY)

@ShiriNmi1520 can u help me if the compose  or the aliyun pull some wrong ,tks

**Comment by JakobLichterfeld** [comment]
You are using `docker-compose` which is deprecated and not supported. See hint https://docs.teslamate.org/docs/maintenance/backup_restore

> If you are using docker-compose, you are using Docker Compose v1, which has been deprecated. Docker Compose commands refer to Docker Compose v2. Consider upgrading your docker setup, see [Migrate to Compose V2](https://docs.docker.com/compose/migrate/)

**Reply by Dalishimin** [reply]
i don't know why.
it v2 now??

```
/volume1/docker# docker-compose --version
Docker Compose version v2.9.0-6413-g38f6acd

```

**Reply by JakobLichterfeld** [reply]
no, `docker-compose` != `docker compose`

**Reply by Dalishimin** [reply]
Now, it should be the v2 version. I tested 3 commands.
However, when pulling, the same error is still prompted:
` Error response from daemon: Get "https://registry-1.docker.io/v2/": EOF`

Another question is: I am using docker on Synology nas. After upgrading to v2, why do I keep getting the message:
` WARN[0000] /volume1/docker/docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion`

![企业微信截图_1721802664350](https://github.com/user-attachments/assets/3f797742-b152-4b22-9bf3-d0f2fb1ccea4)

**Reply by JakobLichterfeld** [reply]
> Error response from daemon: Get "https://registry-1.docker.io/v2/": EOF

docker.io indicates the Linux packages, not the docker container registry, therefore not a TeslaMate issue. Check your DNS, proxy, firewall, and region specific restriction.

> Another question is: I am using docker on Synology nas. After upgrading to v2, why do I keep getting the message:
> `WARN[0000] /volume1/docker/docker-compose.yml: the attribute`version` is obsolete, it will be ignored, please remove it to avoid potential confusion`

You use in your `docker-compose.yml` the version tag at the beginning (for example `version: 3`), which is obsolete and can safely be deleted. See example `docker-compose.yml` here https://docs.teslamate.org/docs/installation/docker/

**Reply by Dalishimin** [reply]
maybe like @ShiriNmi1520 say:

> Maybe it's because the government demands that Docker registry access be [shut down](https://medium.com/@PlanB./chinese-docker-hub-complete-shutdown-how-far-can-kubernetes-image-repositories-go-826706007b8e). It looks like it's even affecting proxy servers.
> For your reference, here's the [post](https://juejin.cn/post/7381760851986120743?searchId=20240702114502657BF3E298100164551D) from JueJin to help you migrate the registry image to Aliyun

i tried ,but failed

**Reply by Dalishimin** [reply]
now ,other way,the last step ,trouble me,failed like this:

![image](https://github.com/user-attachments/assets/53a79d40-c69c-4237-b10a-b50827c04f0a)
